micov/cli.py

```diff
@@ -103,27 +104,45 @@ def compress(data, output, disable_compression, lengths, taxonomy):
         if taxonomy is not None:
             taxonomy = parse_taxonomy(taxonomy)
 
-    # compress data in blocks to avoid loading full mapping data into memory
-    # and compress as we go along.
-    df = compress_from_stream(data, disable_compression=disable_compression)
-    if df is None or len(df) == 0:
-        click.echo("File appears empty...", err=True)
-        sys.exit(0)
-
-    if lengths is None:
-        df.write_csv(output, separator='\t', include_header=True)
+    if os.path.isdir(data):
+        file_list = glob(data + "/*.sam")
+        file_list_gz = glob(data + "/*.sam.gz")
+        file_list_xz = glob(data + "/*.sam.xz")
+        file_list = file_list + file_list_gz + file_list_xz
+
+        dfs = []
+        for samfile in file_list:
+            df = compress_from_stream(samfile,
+                                      disable_compression=disable_compression)
+            if df is None or len(df) == 0:
+                click.echo("File appears empty...", err=True)
+                sys.exit(0)
+            dfs.append(df)
+        coverage = _single_df(_check_and_compress(dfs, compress_size=0))
+        genome_coverage = coverage_percent(coverage, lengths).collect()
     else:
-        genome_coverage = coverage_percent(df, lengths).collect()
-
-        if taxonomy is None:
-            genome_coverage.write_csv(output, separator='\t', include_header=True)
+        # compress data in blocks to avoid loading full mapping data into
```

Comment:
It looks like there is replicated code here. Isn't processing a single file just a special case of processing a directory of files? (i.e., in the single file case, why not iterate over `file_list = [the_file, ]`)